docs/api/js/utils.md

```diff
@@ -1,68 +1,772 @@
+- `provider`: ethers `Provider` object.
+
+#### Outputs
+
+An ABI-encoded array of:
+
+- `nameBytes`: `bytes` object representation of token name.
+- `symbolBytes`: `bytes` object representation of token symbol.
+- `decimalBytes`: `bytes` object representation of token decimal representation.
+
+```ts
+async function getERC20GettersData(l1TokenAddress: string, provider: ethers.providers.Provider): Promise<string> {
+    const token = IERC20MetadataFactory.connect(l1TokenAddress, provider);
+
+    const name = await token.name();
+    const symbol = await token.symbol();
+    const decimals = await token.decimals();
+
+    const coder = new AbiCoder();
+
+    const nameBytes = coder.encode(['string'], [name]);
+    const symbolBytes = coder.encode(['string'], [symbol]);
+    const decimalsBytes = coder.encode(['uint256'], [decimals]);
+
+    return coder.encode(['bytes', 'bytes', 'bytes'], [nameBytes, symbolBytes, decimalsBytes]);
+}
+```
+
+### `getL2HashFromPriorityOp`
+
+Returns the hash of the L2 priority operation from a given transaction receipt and L2 address.
+
+#### Inputs
+
+- `txReceipt`: receipt of the L1 transaction as ethers `TransactionReceipt` object.
+- `zkSyncAddress`: address of sender/transaction ?? on L2 as `Address` object.
+
+```ts
+export function getL2HashFromPriorityOp(
+    txReceipt: ethers.providers.TransactionReceipt,
+    zkSyncAddress: Address
+): string {
+    let txHash: string = null;
+    for (const log of txReceipt.logs) {
+        if (log.address.toLowerCase() != zkSyncAddress.toLowerCase()) {
+            continue;
+        }
+
+        try {
+            const priorityQueueLog = ZKSYNC_MAIN_ABI.parseLog(log);
+            if (priorityQueueLog && priorityQueueLog.args.txHash != null) {
+                txHash = priorityQueueLog.args.txHash;
+            }
+        } catch {}
+    }
+    if (!txHash) {
+        throw new Error('Failed to parse tx logs');
+    }
+
+    return txHash;
+}
+```
+
+### `getHashedL2ToL1Msg`
+
+Returns a keccak encoded message with a given sender address and block number from the L1 messenger contract.
+
+#### Inputs
+
+- `sender`: sender address on L1 ?? as `Address` object.
+- `msg`: encoded message as `BytesLike` object.
+- `txNumberInBlock`: index of the transaction in the block as number.
+
+```ts
+export function getHashedL2ToL1Msg(sender: Address, msg: BytesLike, txNumberInBlock: number) {
+    const encodedMsg = new Uint8Array([
+        0, // l2ShardId
+        1, // isService
+        ...ethers.utils.zeroPad(ethers.utils.hexlify(txNumberInBlock), 2),
+        ...ethers.utils.arrayify(L1_MESSENGER_ADDRESS),
+        ...ethers.utils.zeroPad(sender, 32),
+        ...ethers.utils.arrayify(ethers.utils.keccak256(msg))
+    ]);
+
+    return ethers.utils.keccak256(encodedMsg);
+}
+```
+
+### `getSignature`
+
+Returns the signature of any transaction as a `Uint8Array`.
+
+#### Inputs
+
+- `transaction`: transaction address ?? as ??.
```

Comment:
Let's just not document this method, it is needed only internally. It is not a good thing that it is exposed to the external users, we should  remove it from the public API.

Reply:
It is called from elsewhere in the code. Any interested user will end up going to git to look it up where it is published, ie. it is already public. Are you sure?

Reply:
removed